### PR TITLE
Export layout to 3D modeler with keeping net name speed and robustness improvement

### PR DESCRIPTION
**Pull request by svandenb-dev**
Speed and robustness improvement with export layout design into HFSS and Q3D modeler. 
- added iterative process to scan geometries and retrieve accurately 1 point per primitives
- speed improvement due to single point returned instead on multiple ones with previous method. Speed improvement might be significant on complex design.

### CODE REVIEW

**Review by Samuelopez-ansys — APPROVED** [review]
LGTM

**Review by MaxJPRey — APPROVED** [review]
LGTM.

File: pyaedt/modules/SolveSetup.py [hunk 1]
```diff
@@ -1380,9 +1385,12 @@ def export_to_hfss(self, file_fullname, keep_net_name=False):
...
         succeeded = self._check_export_log(info_messages, error_messages, file_fullname)
         if succeeded and keep_net_name:
-            from pyaedt import Hfss
+            if not is_ironpython:
+                from pyaedt import Hfss
 
-            self._get_net_names(Hfss, file_fullname)
+                self._get_net_names(Hfss, file_fullname)
+            else:
+                self.p_app.odesktop.SetMessages("Exporting layout with keeping net name not supported with ironpython")
```
[the remark this answers is not in the document]
Comment (PipKat) [reply pyaedt/modules/SolveSetup.py:1393]: Messages should end with punctuation.

File: pyaedt/modules/SolveSetup.py [hunk 2]
```diff
@@ -1450,16 +1458,40 @@ def _get_primitives_points_per_net(self):
         primitive_dict = {}
         for net, primitives in net_primitives.items():
             primitive_dict[net] = []
-            if primitives:
+            n = 0
+            while len(primitive_dict[net]) < len(net_primitives[net]):
+                if n > 1000:
```
[the remark this answers is not in the document]
Comment (svandenb-dev) [reply pyaedt/modules/SolveSetup.py]: Done. 1000 as maximum value added to prevent infinite loop in case.

**Review by Samuelopez-ansys — APPROVED** [review]

**Review by maxcapodi78 — APPROVED** [review]

**Pull request merged**